tests/include/mir_test_framework/stub_input_platform.h

```diff
@@ -49,18 +60,16 @@ class StubInputPlatform : public mir::input::Platform
     void pause_for_config() override;
     void continue_after_config() override;
 
-    static void add(std::shared_ptr<mir::input::InputDevice> const& dev);
-    static void remove(std::shared_ptr<mir::input::InputDevice> const& dev);
-    static void register_dispatchable(std::shared_ptr<mir::dispatch::Dispatchable> const& queue);
-    static void unregister_dispatchable(std::shared_ptr<mir::dispatch::Dispatchable> const& queue);
+    void add(std::shared_ptr<mir::input::InputDevice> const& dev);
+    void remove(std::shared_ptr<mir::input::InputDevice> const& dev);
+    void register_dispatchable(std::shared_ptr<mir::dispatch::Dispatchable> const& queue);
+    void unregister_dispatchable(std::shared_ptr<mir::dispatch::Dispatchable> const& queue);
 
 private:
     std::shared_ptr<mir::dispatch::MultiplexingDispatchable> const platform_dispatchable;
     std::shared_ptr<mir::dispatch::ActionQueue> const platform_queue;
     std::shared_ptr<mir::input::InputDeviceRegistry> const registry;
-    static std::atomic<StubInputPlatform*> stub_input_platform;
-    static std::vector<std::weak_ptr<mir::input::InputDevice>> device_store;
-    static std::mutex device_store_guard;
+    std::shared_ptr<DeviceStore> const device_store;
```

Comment:
Would an alternative design be to move these static data members to namespace scope in the implementation file?

Reply:
Good question. Let me know if I'm right in my understanding as to why that won't work. 

The problem there would be that we'd have two symbols included from different sources. The test code still needs to access the `StubInputPlatform` symbol _and_ it needs to load the dynamic library at runtime. Hence, if we included a `static` variable in the `StubInputPlatform`, we would be declaring it twice.

Does that check out for you too?

Reply:
Which `StubInputPlatform` symbol?

Moving the statics means there is, for example, no `StubInputPlatform::stub_input_platform` symbol. Just a local symbol in the implementation file that can't be accessed from elsewhere

Reply:
Hmm I guess you're right there! It should be okay, I will try that

Reply:
Ah nevermind. What about the static methods:

```c++
static void add(std::shared_ptr<mir::input::InputDevice> const& dev);
    static void remove(std::shared_ptr<mir::input::InputDevice> const& dev);
    static void register_dispatchable(std::shared_ptr<mir::dispatch::Dispatchable> const& queue);
    static void unregister_dispatchable(std::shared_ptr<mir::dispatch::Dispatchable> const& queue);
```

Those are public. I needed to move them into the `StubInputPlatformAccessor` to allow those who need it to include those symbols.

Reply:
Remember, you don't need to change this, I was simply curious why this way